src/McRefCommon.cpp

```diff
@@ -18,6 +19,11 @@ double calculateCoalStats(double *elapsed_times, int *num_lineages, int size) {
   return result;
 }
 
+int isLeafNode(int nodeId, int gen) { // TODO - ask Ilan for a better way to answer this question
+  LikelihoodNode *node = getNode(nodeId, gen);
+  return areAlmostEqual(node->age, 0.0);
+}
+
 int isLeafPop(int pop) {
```

Comment:
I think that genealogy leaves are given constant ids in [0 numSamples-1]. Please check.